### PR TITLE
Search language option and searching from different years

### PR DESCRIPTION
**Pull request by mkaflowski**
**Feature: downloading logos from tmdb:**
Priority:
- fanart
- tmdb from your language
- general tmdb logo (without language)
-  tmdb from "en"

**Feature: search language option added**
This is useful if you have other language than "en". For example I hava "pl" and for movie "My Son 2021" I will get wrong result because the right one has polish translation. 
I have added new string but is missing for many language (dont know how to add it fast).

**Change: searching movies from differnt years if not found**
I have file "Nic Smiesznego 1995" for example. But the movie is from 1996 and will not be found. Now it is tried to find from +/- 1 year and without year if not found.

### CODE REVIEW

**Comment by gade01** [comment]
`en_gb` is the source language file instead of `en_us`

https://github.com/xbmc/metadata.themoviedb.org.python/tree/master/resources/language/resource.language.en_gb

**Comment by mkaflowski** [comment]
> `en_gb` is the source language file instead of `en_us`
> 
> https://github.com/xbmc/metadata.themoviedb.org.python/tree/master/resources/language/resource.language.en_gb

It's updated now.

File: python/lib/tmdbscraper/tmdb.py [hunk 1]
```diff
@@ -1,11 +1,13 @@
 from datetime import datetime, timedelta
 from . import tmdbapi
+import xbmc
```
Comment (rmrector) [inline python/lib/tmdbscraper/tmdb.py:3]: Importing `xbmc` is unnecessary.

File: python/scraper.py [hunk 2]
```diff
@@ -23,12 +23,22 @@ def log(msg, level=xbmc.LOGDEBUG):
...
     language = settings.getSettingString('language')
     certcountry = settings.getSettingString('tmdbcertcountry')
-    return TMDBMovieScraper(ADDON_SETTINGS, language, certcountry)
+    search_language = settings.getSettingString('searchlanguage')
+    return TMDBMovieScraper(ADDON_SETTINGS, language, search_language, certcountry)
 
 def search_for_movie(title, year, handle, settings):
     log("Find movie with title '{title}' from year '{year}'".format(title=title, year=year), xbmc.LOGINFO)
     title = _strip_trailing_article(title)
     search_results = get_tmdb_scraper(settings).search(title, year)
```
Comment (rmrector) [inline python/scraper.py:32]: Save the scraper instance returned from `get_tmdb_scraper` here, and then reuse it for the following searches.

File: python/scraper.py [hunk 3]
```diff
@@ -23,12 +23,22 @@ def log(msg, level=xbmc.LOGDEBUG):
...
+
+    if not search_results:
```
Comment (rmrector) [inline python/scraper.py:34]: All three of these are only different from the initial search if year is not None, so flip the ifs like

```py
if year is not None:
    if not search_results:
        search_results = get_tmdb_scraper(settings).search(title,str(int(year)-1))
    if not search_results:
        search_results = get_tmdb_scraper(settings).search(title,str(int(year)+1))
    if not search_results:
        search_results = get_tmdb_scraper(settings).search(title)
```

**Comment by croneter** [comment]
I think you need to change the add-ons settings as well and add the new setting you're using here. See e.g. https://github.com/croneter/PlexKodiConnect/blob/master/resources/settings.xml

**Comment by mkaflowski** [comment]
> I think you need to change the add-ons settings as well and add the new setting you're using here. See e.g. https://github.com/croneter/PlexKodiConnect/blob/master/resources/settings.xml
Version 1.4.3 should work well because I have changed arguments order and set default search language value.

File: python/scraper.py [hunk 4]
```diff
@@ -23,12 +23,22 @@ def log(msg, level=xbmc.LOGDEBUG):
 def get_tmdb_scraper(settings):
     language = settings.getSettingString('language')
     certcountry = settings.getSettingString('tmdbcertcountry')
-    return TMDBMovieScraper(ADDON_SETTINGS, language, certcountry)
+    search_language = settings.getSettingString('searchlanguage')
```
Comment (rmrector) [inline python/scraper.py:26]: croneter is correct, this change also need a new setting in [the settings.xml file](https://github.com/xbmc/metadata.themoviedb.org.python/blob/master/resources/settings.xml#L8). You can copy this line and change `label="30002"` attribute value to your new language string **30013** and the `id="language"` attribute value to **searchlanguage** to match this line.

File: resources/settings.xml [hunk 5]
```diff
@@ -7,6 +7,7 @@
 		<setting label="30004" type="bool" id="trailer" default="true"/>
 		<setting label="30002" type="select" values="ar-AE|ar-SA|be-BY|bg-BG|bn-BD|ca-ES|ch-GU|cs-CZ|da-DK|de-DE|el-GR|en-US|eo-EO|es-ES|es-MX|et-EE|eu-ES|fa-IR|fi-FI|fr-CA|fr-FR|gl-ES|he-IL|hi-IN|hr-HR|hu-HU|id-ID|it-IT|ja-JP|ka-GE|ko-KR|lt-LT|lv-LV|ml-IN|nb-NO|nl-NL|no-NO|pl-PL|pt-BR|pt-PT|ro-RO|ru-RU|sk-SK|sl-SI|sr-RS|sv-SE|ta-IN|th-TH|tr-TR|uk-UA|vi-VN|zh-CN|zh-HK|zh-TW" id="language" default="en-US"/>
 		<setting label="30006" type="select" values="au|bg|br|by|ca|cz|ge|de|dk|ee|es|fi|fr|gb|gr|hr|hu|id|il|in|it|ir|jp|kr|lt|lv|mx|nl|no|pl|pt|ru|si|sv|th|tr|ua|us|vn|zh" id="tmdbcertcountry" default="us"/>
+		<setting label="30013" type="select" values="au|bg|br|by|ca|cz|ge|de|dk|ee|es|fi|fr|gb|gr|hr|hu|id|il|in|it|ir|jp|kr|lt|lv|mx|nl|no|pl|pt|ru|si|sv|th|tr|ua|us|vn|zh" id="tmdbcertcountry" default="us"/>
```
Comment (rmrector) [inline resources/settings.xml:10]: Close. Try this, it's a copy of the original "language" selection dropdown.

```suggestion
		<setting label="30013" type="select" values="ar-AE|ar-SA|be-BY|bg-BG|bn-BD|ca-ES|ch-GU|cs-CZ|da-DK|de-DE|el-GR|en-US|eo-EO|es-ES|es-MX|et-EE|eu-ES|fa-IR|fi-FI|fr-CA|fr-FR|gl-ES|he-IL|hi-IN|hr-HR|hu-HU|id-ID|it-IT|ja-JP|ka-GE|ko-KR|lt-LT|lv-LV|ml-IN|nb-NO|nl-NL|no-NO|pl-PL|pt-BR|pt-PT|ro-RO|ru-RU|sk-SK|sl-SI|sr-RS|sv-SE|ta-IN|th-TH|tr-TR|uk-UA|vi-VN|zh-CN|zh-HK|zh-TW" id="searchlanguage" default="en-US"/>
```

**Comment by rmrector** [comment]
Cheers.

**Comment by KarellenX** [comment]
@mkaflowski 

This, or one of your other PR's, has broken the scraper for some users.
Could you please check the log in this thread for the errors... https://forum.kodi.tv/showthread.php?tid=365037

Happy to discuss with you in that thread.

The complaints are slowly starting to trickle in.

Thanks :)

**Pull request merged**